src/Stratis.Bitcoin/Utilities/Extensions/DBreezeExtensions.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Threading.Tasks;
+using DBreeze;
+using DBreeze.Exceptions;
+using DBreeze.Transactions;
+using Microsoft.Extensions.Logging;
+
+namespace Stratis.Bitcoin.Utilities.Extensions
+{
+    public class RetryOptions
+    {
+        public RetryOptions()
+        {
+            this.RetryCount = 1;
+            this.Delay = TimeSpan.FromMilliseconds(100);
+        }
+
+        public RetryOptions(short retryCount, TimeSpan delay)
+        {
+            if (retryCount < 1)
+                throw new ArgumentOutOfRangeException(nameof(retryCount), "Retry count cannot be less or equal to 0");
+
+            this.RetryCount = retryCount;
+            this.Delay = delay;
+        }
+
+        public static RetryOptions Default => new RetryOptions();
+
+        public short RetryCount { get; }
+
+        public TimeSpan Delay { get; }
+
+        public override string ToString()
+        {
+            return $"Retry count: {this.RetryCount}; Delay: {this.Delay}";
+        }
+    }
+
+    public static class DBreezeExtensions
+    {
+        public static void TransactionalCommit(this DBreezeEngine dbreeze, ILogger logger, Action<Transaction> handler, RetryOptions retryOptions = null)
+        {
+            if (retryOptions == null || retryOptions.RetryCount == 0)
+            {
+                ExecuteTransaction(dbreeze, handler);
+                return;
+            }
+
+            for (int i = 0; i <= retryOptions.RetryCount; i++)
+            {
+                try
+                {
+                    ExecuteTransaction(dbreeze, handler);
+                    return;
```

Comment:
The return is redundant

Reply:
It is not redundant, we want to exit out of for loop either using return or break.